File: app/src/main/java/de/sysitprep/app/ui/home/HomeViewModel.kt
package de.sysitprep.app.ui.home

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import de.sysitprep.app.ExamForgeApp
import de.sysitprep.app.data.model.UserStats
import kotlinx.coroutines.launch

class HomeViewModel(application: Application) : AndroidViewModel(application) {

    private val repo = (application as ExamForgeApp).repository

    private val _stats = MutableLiveData<UserStats>()
    val stats: LiveData<UserStats> = _stats

    private val _bookmarkCount = MutableLiveData<Int>()
    val bookmarkCount: LiveData<Int> = _bookmarkCount

    init {
        loadStats()
        observeBookmarks()
    }

    private fun loadStats() {
        viewModelScope.launch {
            val stats = repo.getUserStats()
            _stats.value = stats
        }
    }

    private fun observeBookmarks() {
        viewModelScope.launch {
            repo.getBookmarkedQuestions().collect { bookmarks ->
                _bookmarkCount.value = bookmarks.size
            }
        }
    }

    fun refreshStats() = loadStats()
}

File: ExamForge-iOS/ExamForge/ViewModels/HomeViewModel.swift
import Foundation
import Combine

@MainActor
final class HomeViewModel: ObservableObject {
    @Published var stats = UserStats()
    @Published var bookmarkCount = 0

    private let repository: QuestionRepository

    init(repository: QuestionRepository) {
        self.repository = repository
    }

    func refreshStats() {
        Task {
            stats = await repository.getUserStats()
            let bookmarks = await repository.getBookmarkedQuestions()
            bookmarkCount = bookmarks.count
        }
    }
}
